sidechain/consensus/common/src/block_import_queue_worker.rs

```diff
@@ -106,33 +105,32 @@ impl<ParentchainBlock, SignedSidechainBlock, BlockImportQueue, PeerBlockSyncer>
 		);
 
 		if let Ok(candidates) = self.block_import_queue.pop_all() {
-			let mut sorted_candidates = candidates
+			number_of_imported_blocks = candidates
 				.iter()
 				.map(|b| (b.block().header().block_number(), b))
-				.collect::<Vec<(SidechainBlockNumber, &SignedSidechainBlock)>>();
-			sorted_candidates.sort_by_key(|a| a.0);
-			number_of_imported_blocks = sorted_candidates
-				.iter()
+				.sorted_by_key(|a| a.0)
 				.group_by(|&a| a.0)
 				.into_iter()
 				.filter_map(|(block_number, competitors)| {
 					let mut competitors: Vec<&SignedSidechainBlock> =
-						competitors.map(|&c| c.1).collect();
+						competitors.map(|c| c.1).collect();
 					// deterministic import order decreases chances for forks
 					competitors.sort_by_key(|a| a.block().hash());
 					trace!("nr of competitors for block {}: {}", block_number, competitors.len());
```

Comment:
Unfortunately, we can only know the size of the iterator if we collected, so I guess, we can't omit this part of the code